src/mpid/ch4/src/ch4_recv.h

```diff
@@ -136,19 +139,32 @@ MPL_STATIC_INLINE_PREFIX int MPID_Recv_init(void *buf,
             MPIR_ERR_POP(mpi_errno);
         }
 
-        mpi_errno =
-            MPIDI_NM_mpi_recv_init(buf, count, datatype, rank, tag, comm, context_offset, av,
-                                   &(MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(*request)));
+        MPIDI_CH4I_REQUEST(*request, is_local) = 1;
 
-        if (mpi_errno != MPI_SUCCESS) {
-            MPIR_ERR_POP(mpi_errno);
-        }
+        if (!MPIR_Request_is_complete(*request) && !MPIDI_CH4U_REQUEST_IN_PROGRESS(*request)) {
```

Comment:
From #3339 by @hajimefu:

> Also, is checking for `MPIR_Request_is_complete(*request)` valid here? Here `*request` is an inactive persistent request, which is always considered as "completed" if I understand correctly.

I think this might be a valid point. Perhaps this should drop the completion check and just look at whether or not it's in progress.

Reply:
Removed the completion check.

Reply:
MPIDI_CH4U_REQUEST_IN_PROGRESS caused segmentation fault when request is completed (due to null pointer dereference inside MPIDI_CH4U_REQUEST_IN_PROGRESS).

To avoid segmentation fault, in the first we must check MPIR_Request_is_complete() and if that is not true, we can check MPIDI_CH4U_REQUEST_IN_PROGRESS().